polkadot/node/core/backing/src/lib.rs

```diff
@@ -1010,21 +1026,50 @@ async fn construct_per_relay_parent_state<Context>(
 	let minimum_backing_votes =
 		try_runtime_api!(request_min_backing_votes(parent, session_index, ctx.sender()).await);
 
+	// TODO: https://github.com/paritytech/polkadot-sdk/issues/1940
+	// Once runtime ver `DISABLED_VALIDATORS_RUNTIME_REQUIREMENT` is released remove this `if`
+	// statement, add `request_from_runtime` call to the `try_join!` call above and use
+	// `try_runtime_api!` to get `disabled_validators`
+	let disabled_validators = if has_required_runtime(
+		ctx.sender(),
+		parent,
+		RuntimeApiRequest::DISABLED_VALIDATORS_RUNTIME_REQUIREMENT,
+	)
+	.await
+	{
+		let disabled_validators = request_from_runtime(parent, ctx.sender(), |tx| {
+			RuntimeApiRequest::DisabledValidators(tx)
+		})
+		.await
+		.await
+		.map_err(Error::JoinMultiple)?;
+
+		let disabled_validators = try_runtime_api!(disabled_validators);
+		disabled_validators
+	} else {
+		gum::debug!(target: LOG_TARGET, "Runtime doesn't support `DisabledValidators` - continuing with an empty disabled validators set");
+		vec![]
+	};
+
```

Comment:
This code needs to stay until we release v8: https://github.com/paritytech/polkadot-sdk/issues/1940

Reply:
can we extract that into subsystem-util or somewhere where statement-distribution and dispute-coordinator can reuse it?

Reply:
Yes, I was thinking about the same and I started working on it but against master.
My idea is to have a wrapper around `request_from_runtime` which does a version check and optionally returns a fallback version. I'll open a PR against master because I think this will be useful in general. We can backport it here easily after that.

Reply:
I went into a macro rabbit hole. I'll just extract this implementation here and finish the other version in a more appropriate time.

Reply:
Sure, sounds good. However, I didn't mean to use a macro for that, just a regular function should be OK.